tests/unit-tests/test-class-grading.php

```diff
@@ -183,6 +183,23 @@ public function testGradeGapFillQuestionRegEx( $answer, $found, $not_found, $cas
 			$response = Sensei_Grading::grade_gap_fill_question( $question_id, $not_found_item );
 			$this->assertFalse( $response, "Expecting {$not_found_item} to not match {$answer}" );
 		}
+
+	}
+
+	/**
+	 *
+	 * This tests generated graded lessons and makes sure that the function
+	 * get graded lessons average is returning expected value for average lesson.
+	 *
+	 * @covers Sensei_Grading::get_graded_lessons_average_grade
+	 * @since 4.2.0
+	 */
+	public function testGetGradedLessonsAverage() {
+
+		$this->factory->generate_graded_lessons( 5 );
+		$graded_lessons_average_grade = Sensei_Grading::get_graded_lessons_average_grade();
+
+		$this->assertEquals( $graded_lessons_average_grade, 30 );
```

Comment:
It's not clear to me why the expected value is 30 without looking at how `generate_graded_lessons` is implemented. Perhaps that function should also take a grade value so that it's more obvious?

Reply:
well the grade values are generated based on the how many of them I want, but I can just send the values instead, it does make it more clear.

Reply:
Changed graded tests [in this commit](https://github.com/Automattic/sensei/pull/4865/commits/72f12f6213e428b3dfc4a029295cf430ff5966ec)